src/python/WMComponent/DBS3Buffer/DBSBufferBlock.py

```diff
@@ -232,10 +232,19 @@ def setProcessingVer(self, procVer):
         Set the block's processing version.
         """
         # compatibility statement for old style proc ver (still needed ?)
-        if procVer.count("-") == 1:
-            self.data["processing_era"]["processing_version"] = procVer.split("-v")[1]
+        pver = procVer
+        if isinstance(pver, str) or isinstance(pver, bytes):
+            if procVer.count("-") == 1:
+                pver = procVer.split("-v")[1]
+            procVer = int(pver)
+        elif isinstance(pver, int):
+            procVer = pver
+        elif pver == None:
```

Comment:
Here you change the behavior. I'd suggest not to default to anything other than what comes from the database.

Reply:
Actually, I suggest to simply assign the value that it receives from the database (thus keeping basically the same logic, without the ancient compatibility code). I'm going to suggest a schema change - later in this review - that will ensure the correct data type.